src/components/forms/TextInput/TextInput.tsx

```diff
@@ -6,21 +6,33 @@ import {
   castNullableStringValue,
   WithNullableValue,
 } from '../../../utils/react/nullableValue';
+import { useFieldProps } from '../Form';
 
 import { CubeTextInputBaseProps, TextInputBase } from './TextInputBase';
 
 export type CubeTextInputProps = WithNullableValue<CubeTextInputBaseProps>;
 
+/**
+ * TextInputs are text inputs that allow users to input custom text entries
+ * with a keyboard. Various decorations can be displayed around the field to
+ * communicate the entry requirements.
+ */
 export const TextInput = forwardRef(function TextInput(
   props: CubeTextInputProps,
   ref: ForwardedRef<HTMLInputElement>,
 ) {
-  castNullableStringValue(props);
-
+  props = castNullableStringValue(props);
   props = useProviderProps(props);
+  props = useFieldProps(props, {
+    defaultValidationTrigger: 'onBlur',
+    valuePropsMapper: ({ value, onChange }) => ({
+      onChange,
+      value: value?.toString() ?? '',
```

Comment:
Did you mean `value?.toString?.() ?? '',`?

Reply:
Not sure we need to add optional chaining to `toString` since I can't remember a non `undefined` type that doesn't have `toString()` implemented

Reply:
Bad example, but `request.body` doesn't have.
It's not so difficult to create one: `Object.create(null)`.
But yes, we can leave it as is without `?`.